python/paddle/fluid/dygraph/layers.py

```diff
@@ -1426,11 +1426,28 @@ def transform(t, device, dtype, blocking):
             if dtype is None:
                 dtype = t.dtype
 
-            new_t = t._copy_to(device, blocking)
-            if dtype is not None and dtype != t.dtype:
-                new_t = new_t.cast(dtype=dtype)
+            if isinstance(t, framework.ParamBase):
```

Comment:
the `ParamBase._copy_to` is still error, whether override the `ParamBase._copy_to` method to fix this bug?